walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,96 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.HashSet;
+import java.util.Set;
+
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
+    Collection<TimeRange> BusyTimes= new ArrayList <TimeRange>();
+    Collection<TimeRange> FreeTimes = new ArrayList <TimeRange>();
+    int start_of_day = TimeRange.START_OF_DAY;
+    int eod = TimeRange.END_OF_DAY;
+    int prev_end_time =0 ; // End time of the previous event accessed
+
+    // Inital checks for input validity(no attendees ,time is more than 24 hours)
+    if (request.getAttendees().isEmpty()) {
+            FreeTimes.add(TimeRange.fromStartEnd(start_of_day,eod, true));
+            return FreeTimes;
+    }
+    
+    if( request.getDuration() > 24*60) {
+            return FreeTimes;
+    }
+    
+    for(Event event : events ){
+        for( String attendee : request.getAttendees()){
+            if(event.getAttendees().contains(attendee)){ // Make sure we only check the events of people going and ignore those who are not
+                /* Overlap case one
+                *   [  a  ]  
+                *       [  b  ]
+                * <------WHOLE DAY ------>
+                * If there is a prior meeting a that cuts into b, the busy time recorded for meeting b will only be from the range of (end of a -- end of b)
+                */  
```

Comment:
I really like the comments showing the cases!!!

Reply:
Thanks!